### PR TITLE
scrapy.utils.data_path type hint change

### PR DESCRIPTION
**Pull request by cakemd**
Updating type hint for `scrapy.utils.project.data_path`. 

This is a small change which brings a part of solution of the issue [#5739](https://github.com/scrapy/scrapy/issues/5739).

The type hint change was suggested [here](https://github.com/scrapy/scrapy/issues/5739#issuecomment-1789203848).

### CODE REVIEW

File: scrapy/utils/project.py [hunk 1]
```diff
@@ -44,7 +46,7 @@ def project_data_dir(project: str = "default") -> str:
     return str(d)
 
 
-def data_path(path: str, createdir: bool = False) -> str:
+def data_path(path: Union[str, PathLike, Path], createdir: bool = False) -> str:
```
Comment (wRAR) [inline scrapy/utils/project.py:49]: `pathlib.Path` is a subclass of `os.PathLike` so it's not needed to be mentioned separately here.

Comment (cakemd) [reply scrapy/utils/project.py]: @wRAR updated

Comment (wRAR) [reply scrapy/utils/project.py]: I meant removing `Path`, not removing `PathLike`. `Union[str, PathLike]` is the usual hint for these APIs.

Comment (cakemd) [reply scrapy/utils/project.py]: Oh, right. Updated.

**Comment by wRAR** [comment]
Can you please also install and use pre-commit as described at https://docs.scrapy.org/en/latest/contributing.html#pre-commit ?

**Comment by cakemd** [comment]
@wRAR sure, done. Thanks.

**Review by wRAR — APPROVED** [review]

**Comment by Gallaecio** [comment]
Thanks!

**Pull request merged**